crypto/keys/internal/ecdsa/pubkey.go

```diff
@@ -0,0 +1,83 @@
+package ecdsa
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/sha256"
+	"encoding/asn1"
+	"fmt"
+	"math/big"
+
+	tmcrypto "github.com/tendermint/tendermint/crypto"
+
+	"github.com/cosmos/cosmos-sdk/types/address"
+	"github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// signature holds the r and s values of an ECDSA signature.
+type signature struct {
+	R, S *big.Int
+}
+
+type PubKey struct {
+	ecdsa.PublicKey
+
+	// cache
+	address tmcrypto.Address
+}
+
+// String implements PubKey interface.
+func (pk *PubKey) Address(protoName string) tmcrypto.Address {
```

Comment:
I think these comments are misleading, these are helpers called in the actual implementations of the PubKey interface

Reply:
Yes, I left the from the original implementation (when this structure implemented the PubKey interface). I will update the comment.